test/integration/controller/jobset_controller_test.go

```diff
@@ -745,14 +1305,29 @@ func completeJob(job *batchv1.Job) {
 			Status: corev1.ConditionTrue,
 		}),
 	})
+	gomega.Eventually(func() error {
+		var jobGet batchv1.Job
+		if err := k8sClient.Get(ctx, types.NamespacedName{Name: job.Name, Namespace: job.Namespace}, &jobGet); err != nil {
+			return err
+		}
+		jobGet.Status.Succeeded = ptr.Deref(jobGet.Spec.Parallelism, 0)
+		return k8sClient.Status().Update(ctx, &jobGet)
+	}, timeout, interval).Should(gomega.Succeed())
+
```

Comment:
No need for this block of code, just add `Succeeded = ptr.Deref(job.Spec.Parallelism, 0)` to the above call, like this:


```
updateJobStatus(job, batchv1.JobStatus{
                 Succeeded = ptr.Deref(job.Spec.Parallelism, 0)
		Conditions: append(job.Status.Conditions, batchv1.JobCondition{
			Type:   batchv1.JobComplete,
			Status: corev1.ConditionTrue,
		}),
	})
```